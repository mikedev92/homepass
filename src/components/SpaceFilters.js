// @flow
import React from 'react';
import {
  Header,
  Container,
  Content,
  Button,
  Icon,
  ListItem,
  Text,
  Body,
  Right,
  Left,
  Radio,
  List,
} from 'native-base';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import { compose } from 'recompose';
import { get } from 'lodash';
import { connect } from 'react-redux';
import graphqlLoader from './common/graphqlLoader';
import { type State, spaceSelector } from '../reducers';
import * as fromActions from '../actions/Space';

type Props = {
  data: ApolloData<void>,
  toggleFilter: string => void,
  filters: { [string]: boolean },
  navigation: Navigation,
};
function SpaceFilters({ data, toggleFilter, filters, navigation }: Props) {
  const spaceFilters: Array<SpaceFilter> = get(data, 'viewer.spaceFilters', []);
  return (
    <Container backgroundDark>
      <Header bordered>
        <Left>
          <Button onPress={() => navigation.goBack()} transparent primary>
            <Icon name="arrow-back" />
          </Button>
        </Left>
        <Body />
        <Right />
      </Header>
      <Content>
        <ListItem itemHeader>
          <Text>
            SHOW
          </Text>
        </ListItem>
        <List>
          {spaceFilters.map(
            (filter, i) =>
              filter &&
              <ListItem
                onPress={() => filter.tag && toggleFilter(filter.tag, !filters[filter.tag])}
                key={filter.tag}
                last={i === spaceFilters.length - 1}
              >
                <Text>{filter.label}</Text>
                <Right>
                  <Radio selected={filters[filter.tag || '']} />
                </Right>
              </ListItem>,
          )}
        </List>
      </Content>
    </Container>
  );
}

const SpaceFiltersQuery = gql`
  query {
    viewer {
      spaceFilters {
        label
        tag
      }
    }
  }
`;

const mapStateToProps = (state: State) => ({
  filters: spaceSelector(state).filters,
});
const mapDispatchToProps = (dispatch: Dispatch) => ({
  toggleFilter: (key, value) => dispatch(fromActions.setFilter(key, value)),
});
const connector = connect(mapStateToProps, mapDispatchToProps);

export default compose(graphql(SpaceFiltersQuery), graphqlLoader, connector)(SpaceFilters);
