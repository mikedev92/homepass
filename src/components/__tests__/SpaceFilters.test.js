// @flow
import React from 'react';
import { shallow } from 'enzyme';
import SpaceFilters from '../SpaceFilters';

const testProps = [
  {
    filters: { OPEN: true },
    data: {
      viewer: {
        spaceFilters: [{ tag: 'OPEN', label: 'woo' }],
      },
    },
  },
  {
    filters: { OPEN2: true },
    data: {
      viewer: {
        spaceFilters: [
          { tag: 'OPEN', label: 'woo' },
          { tag: 'OPEN1', label: 'woo 1' },
          { tag: 'OPEN2', label: 'woo 2' },
        ],
      },
    },
  },
];
describe('<SpaceFilters />', () => {
  it('matches snapshots', () => {
    testProps.forEach((props) => {
      const baseProps = {
        toggleFilter: jest.fn(),
      };
      const wrapper = shallow(<SpaceFilters {...baseProps} {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('call set visibility with the correct value', () => {
    const toggleFilter = jest.fn();
    const wrapper = shallow(
      <SpaceFilters
        filters={{}}
        data={{
          viewer: {
            spaceFilters: [{ tag: 'OPEN', label: 'open' }],
          },
        }}
        toggleFilter={toggleFilter}
      />,
    );
    const listItem = wrapper.find({ last: true });
    listItem.props().onPress();
    expect(toggleFilter).toHaveBeenCalledTimes(1);
    expect(toggleFilter).toHaveBeenCalledWith('OPEN', true);
  });
});
