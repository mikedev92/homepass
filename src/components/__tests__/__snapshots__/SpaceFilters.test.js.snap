// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpaceFilters /> matches snapshots 1`] = `
<Styled(Container)
  backgroundDark={true}
>
  <Styled(Header)
    bordered={true}
  >
    <Styled(Left)>
      <Styled(Button)
        onPress={[Function]}
        primary={true}
        transparent={true}
      >
        <Styled(Icon)
          name="arrow-back"
        />
      </Styled(Button)>
    </Styled(Left)>
    <Styled(Body) />
    <Styled(Right) />
  </Styled(Header)>
  <Styled(Content)>
    <Styled(ListItem)
      itemHeader={true}
    >
      <Styled(Text)>
        SHOW
      </Styled(Text)>
    </Styled(ListItem)>
    <Styled(List)>
      <Styled(ListItem)
        last={true}
        onPress={[Function]}
      >
        <Styled(Text)>
          woo
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={true}
          />
        </Styled(Right)>
      </Styled(ListItem)>
    </Styled(List)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`<SpaceFilters /> matches snapshots 2`] = `
<Styled(Container)
  backgroundDark={true}
>
  <Styled(Header)
    bordered={true}
  >
    <Styled(Left)>
      <Styled(Button)
        onPress={[Function]}
        primary={true}
        transparent={true}
      >
        <Styled(Icon)
          name="arrow-back"
        />
      </Styled(Button)>
    </Styled(Left)>
    <Styled(Body) />
    <Styled(Right) />
  </Styled(Header)>
  <Styled(Content)>
    <Styled(ListItem)
      itemHeader={true}
    >
      <Styled(Text)>
        SHOW
      </Styled(Text)>
    </Styled(ListItem)>
    <Styled(List)>
      <Styled(ListItem)
        last={false}
        onPress={[Function]}
      >
        <Styled(Text)>
          woo
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio) />
        </Styled(Right)>
      </Styled(ListItem)>
      <Styled(ListItem)
        last={false}
        onPress={[Function]}
      >
        <Styled(Text)>
          woo 1
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio) />
        </Styled(Right)>
      </Styled(ListItem)>
      <Styled(ListItem)
        last={true}
        onPress={[Function]}
      >
        <Styled(Text)>
          woo 2
        </Styled(Text)>
        <Styled(Right)>
          <Styled(Radio)
            selected={true}
          />
        </Styled(Right)>
      </Styled(ListItem)>
    </Styled(List)>
  </Styled(Content)>
</Styled(Container)>
`;
